lib/grid_view.rb

```diff
@@ -1,17 +1,34 @@
+require_relative "coordinates"
+
 class GridView
-  attr_reader :output
+  attr_reader :grid, :output
 
-  def initialize(output:)
+  def initialize(grid:, output:)
+    @grid = grid
     @output = output
   end
 
   def render
-    output.puts(template)
+    output.puts(rendered_template)
   end
 
   private
 
+  def rendered_template
+    template % all_cells
+  end
+
   def template
     @template ||= File.read("lib/grid.txt")
   end
+
+  def all_cells
```

Comment:
Even though that style of string interpolation was easy and fast, the way that it necessitates switching out of OO and into raw data is a bummer. I really don't like seeing the `to_h` and then the construction of an array.  But it makes the template system go. 

I think if you wanted to dive deeper into the OO, think about how you could create a template system that didn't require switching from OO to data concerns.